### PR TITLE
Graph building

### PR DESCRIPTION
**Pull request by mockingbird2**
Bootstrapping first community analysis and introduce new pipeline tools

### CODE REVIEW

**Comment by WGierke** [comment]
Is it possible to plot the resulting graph and persist it in the notebook?

**Comment by mockingbird2** [comment]
@WGierke Do you want to persist the graph for loading it later or do you just want to see a representation (plot)? I am currently not sure what you mean with "persist in the notebook".

**Comment by WGierke** [comment]
I mean the plot.

⁣Sent from my mobile. Please excuse the brevity.​

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ data/raw/
 
 # Virtual Environment
 venv*
+src/python-igraph
```
Comment (WGierke) [inline .gitignore]: Why is that necessary?

File: notebooks/1.4-fp-graph-communities.ipynb [hunk 2]
```diff
@@ -0,0 +1,2348 @@
...
+       "<path style=\"fill:none;stroke-width:1;stroke-linecap:butt;stroke-linejoin:miter;stroke:rgb(26.666667%,26.666667%,26.666667%);stroke-opacity:1;stroke-miterlimit:10;\" d=\"M 246.953125 247.292969 L 384.78125 60.925781 \"/>\n",
+       "<path style=\"fill:none;stroke-width:1;stroke-linecap:butt;stroke-linejoin:miter;stroke:rgb(26.666667%,26.666667%,26.666667%);stroke-opacity:1;stroke-miterlimit:10;\" d=\"M 319.300781 433.277344 C 297.367188 449.484375 286.300781 471.222656 286.105469 498.492188 \"/>\n",
+       "<path style=\"fill:none;stroke-width:1;stroke-linecap:butt;stroke-linejoin:miter;stroke:rgb(26.666667%,26.666667%,26.666667%);stroke-opacity:1;stroke-miterlimit:10;\" d=\"M 319.300781 433.277344 C 319.105469 460.550781 308.039062 482.289062 286.105469 498.492188 \"/>\n",
+       "<path style=\"fill:none;stroke-width:1;stroke-linecap:butt;stroke-linejoin:miter;stroke:rgb(26.666667%,26.666667%,26.666667%);stroke-opacity:1;stroke-miterlimit:10;\" d=\"M
```
Comment (WGierke) [inline notebooks/1.4-fp-graph-communities.ipynb]: Is it also possible to visualize this graph? I guess this one already includes knowledge about detected communities?

File: .gitignore (hunk 1 above)
Comment (mockingbird2) [reply .gitignore]: When I install the requirements it is creating this folder and building igraph in this directory.

File: src/filters.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+def use_n_most_occuring_users(df, n):
+    users = df.groupby(df['by']).size()
+    filtered_users, _ = zip(
+        *users.sort_values(ascending=False).head(100).items())
```
Comment (WGierke) [inline src/filters.py]: Can you make the 100 a default parameter?

Comment (mockingbird2) [reply src/filters.py]: This should not look this way. I fix it.

File: notebooks/1.4-fp-graph-communities.ipynb (hunk 2 above)
Comment (mockingbird2) [reply notebooks/1.4-fp-graph-communities.ipynb]: I have not looked into it further since the dendrogram plot was not working for me so far.

File: src/filters.py (hunk 3 above)
Comment (fawind) [reply src/filters.py]: What is the `n` doing?

**Comment by mockingbird2** [comment]
I added the plotted dendrogram and a basic community approximation plot. Unfortunately, an optimal community computation takes more than 16GB of RAM. @WGierke

**Comment by WGierke** [comment]
Good stuff! I think this should be enough to show the supervisor that we're somehow on the right way. Getting a machine with more RAM should also be no problem once we want to create the final plots.
